synapse/lib/cell.py

```diff
@@ -2325,6 +2330,63 @@ def getArgParser(cls, conf=None):
 
         return pars
 
+    @classmethod
+    async def provAhaSvc(cls, opts, conf):
+
+        donepath = s_common.genpath(opts.dirn, 'prov.done')
+        if os.path.isfile(donepath):
+            return
+
+        provurl = conf.get('aha:provision')
+        if provurl is None:
+            return
+
+        certdir = s_certdir.CertDir(path=s_common.gendir(opts.dirn, 'certs'))
+
+        confedit = {}
+        async with await s_telepath.openurl(provurl) as prov:
+
+            certdir.saveCaCertByts(await prov.getCaCert())
+
+            provinfo = await prov.getProvInfo()
+
+            name = provinfo.get('name')
```

Comment:
could provinfo include aha:leader

Reply:
If we wanna go with the convention that any `.` in the provname means you're being provisioned as a member of a cluster, yep!